src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+import { Box, Typography, useTheme, Link } from '@mui/material';
+import React, { useState } from 'react';
+
+import FacebookStatic from './assets/Facebook.svg?url';
+import GithubStatic from './assets/Github.svg?url';
+import GoogleStatic from './assets/Google.svg?url';
+import {
+  getPageContainerStyles,
+  getFormStyles,
+  getLogoStyles,
+  getInputStyles,
+  getButtonStyles,
+  haveAccountTextStyles,
+  loginLinkStyles,
+  signupWithTextStyles,
+  getSocialIconStyles,
+  legalTextStyles,
+  legalLinkStyles,
+} from './SignUpPage.styles';
+
+import type { SignUpPageProps } from './SignUpPage.types';
+
+const FacebookIcon: string | undefined = FacebookStatic ?? undefined;
+const GithubIcon: string | undefined = GithubStatic ?? undefined;
+const GoogleIcon: string | undefined = GoogleStatic ?? undefined;
+
+type LogoFallbackProps = {
+  orientation?: string;
+  width?: string | number;
+  height?: string | number;
+  alt?: string;
+};
+
+type FormInputFallbackProps = {
+  label?: string;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  errorMessage?: string;
+  sx?: Record<string, unknown>;
+  type?: string;
+};
+
+type PrimaryButtonFallbackProps = {
+  children?: React.ReactNode;
+  onClick?: (e: React.FormEvent) => void;
+  disabled?: boolean;
+  loading?: boolean;
+  sx?: Record<string, unknown>;
```

Comment:
PrimaryButtonFallbackProps sx prop is typed as Record<string, unknown>. Use a proper MUI sx type (SxProps<Theme>) instead of a generic record to provide type-safety and better DX.

**Sugerencia:**
```typescript
Use sx?: SxProps<Theme> (imported from '@mui/system' or '@mui/material').
```